unix/x0vncserver/x0vncserver.cxx

```diff
@@ -113,6 +113,18 @@ static void CleanupSignalHandler(int /*sig*/)
   caughtSignal = true;
 }
 
+static void migrateDeprecatedOptions()
+{
+  rfb::VoidParameter* param = Configuration::getParam("ZlibLevel");
+  if (param == NULL)
+    return;
+  
+  int compressLevel = *((rfb::IntParameter*) param);
+  if (compressLevel != -1) {
+    vlog.info("ZlibLevel is deprecated, let clients set the compression level instead.");
+  }
+}
```

Comment:
My mistake, completely forgot about Xvnc.

The encoders are only constructed once a client actually connects to the servers. I figured it would be nice to display a warning message as soon as the server starts instead of when a client connects, so I moved the logging to Configuration.cxx which parses arguments during startup in both x0vncserver and Xvnc.

Reply:
My mistake, completely forgot about Xvnc.

The encoders are only constructed once a client actually connects to the servers. I figured it would be nice to display a warning message as soon as the server starts instead of when a client connects, so I moved the logging to Configuration.cxx which parses arguments during startup in both x0vncserver and Xvnc.

Reply:
Reasonable concern, but I don't think it is worth breaking modularity in `Configuration.cxx` for. I think this is a very rarely used option. So we can live with the fact that the warning isn't perfect to keep the code simple and clean.

If we get many confused users, then we can always revisit the issue again.

Reply:
Fair enough, I've moved the warning to the ZRLEEncoder's constructor.